internal/service/resourcepolicy/model_test.go

```diff
@@ -0,0 +1,171 @@
+package resourcepolicy_test
+
+import (
+	"context"
+	_ "embed"
+	"encoding/json"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/service/resourcepolicy"
+	"github.com/stretchr/testify/assert"
+	"go.mongodb.org/atlas-sdk/v20240805003/admin"
+)
+
+var (
+	//go:embed testdata/policy_clusterForbidCloudProvider.json
+	clusterForbidCloudProviderJSON string
+	//go:embed testdata/policy_clusterForbidCloudProvider_no_name.json
+	clusterForbidCloudProviderNoNameJSON string
+	//go:embed testdata/policy_multipleEntries.json
+	policyMultipleEntriesJSON string
+	userIDCreate              = "65def6f00f722a1507105ad8"
+	userNameCreate            = "mvccpeou"
+	userIDUpdate              = "65def6f00f722a1507105ad9"
+	userNameUpdate            = "updateUser"
+	createdDate               = "2024-09-10T14:59:34Z"
+	lastUpdatedDate           = "2024-09-10T14:59:35Z"
+	orgID                     = "65def6ce0f722a1507105aa5"
+	policyID                  = "66e05ed6680f032312b6b22b"
+	policyBody                = "\n\tforbid (\n\t\tprincipal,\n\t\taction == cloud::Action::\"cluster.createEdit\",\n\t\tresource\n\t) when {\n\t\tcontext.cluster.cloudProviders.containsAny([cloud::cloudProvider::\"aws\"])\n\t};"
+	policies0ID               = "66e05ed6680f032312b6b22a"
+	version                   = "v1"
+)
+
+type sdkToTFModelTestCase struct {
+	SDKRespJSON     string
+	userIDCreate    string
+	userIDUpdate    string
+	userNameCreate  string
+	userNameUpdate  string
+	createdDate     string
+	lastUpdatedDate string
+	orgID           string
+	policyID        string
+	policyBody      string
+	policies0ID     string
+	version         string
+}
+
+func TestResourcePolicySDKToTFModelFull(t *testing.T) {
+	testCases := map[string]sdkToTFModelTestCase{
+		// try no name
+		"clusterForbidCloudProvider": {
+			SDKRespJSON:     clusterForbidCloudProviderJSON,
+			userIDCreate:    userIDCreate,
+			userIDUpdate:    userIDUpdate,
+			userNameCreate:  userNameCreate,
+			userNameUpdate:  userNameUpdate,
+			createdDate:     createdDate,
+			lastUpdatedDate: lastUpdatedDate,
+			orgID:           orgID,
+			policyID:        policyID,
+			policyBody:      policyBody,
+			policies0ID:     policies0ID,
+			version:         version,
+		},
+	}
+
+	for testName, tc := range testCases {
+		t.Run(testName, func(t *testing.T) {
+			SDKModel := parseSDKModel(t, tc.SDKRespJSON)
+			ctx := context.Background()
+			resultModel, diags := resourcepolicy.NewTFResourcePolicyModel(ctx, &SDKModel)
+			assertDiagsOK(t, diags)
+
+			asserter := assert.New(t)
```

Comment:
why is this needed? wouldn't `assert.WhateverMethod()` work?

Reply:
We use this in other places to [ex1](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/CLOUDP-272606_rp_model_conversions/internal/testutil/acc/attribute_checks_test.go#L62), [ex2](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/CLOUDP-272606_rp_model_conversions/internal/testutil/mig/test_case_test.go#L28).
Just makes each of the assert calls a bit simpler